File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> Count;
        vector<int> result;
        for(int num : nums){
            Count[num]++;
        }
        vector<pair<int,int>> array;
        for(const auto&p : Count){
            array.push_back({p.second, p.first});
        }
        sort(array.rbegin(), array.rend());

        for(int i = 0; i < k; i++){
            result.push_back(array[i].second);
        }
        return result;
    }
};

File: Data Structures & Algorithms/boats-to-save-people/submission-0.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        n = len(people)
        l, r = 0, n - 1
        count = 0
        people.sort()
        while(l <= r):
            if l == r:
                count += 1
                break
            
            if people[l] + people[r] <= limit:
                l += 1
            
            r -= 1
            count += 1
        return count